src/sagemaker/jumpstart/artifacts/image_uris.py

```diff
@@ -118,10 +118,35 @@ def _retrieve_image_uri(
     )
 
     if image_scope == JumpStartScriptScope.INFERENCE:
```

Comment:
nit: having a `_retrieve_image_for_instance_type()` and/or a `_retrieve_image_for_ecr_specs()` would help overall code clarity